### PR TITLE
Manufacturing dataproducts

### PR DESCRIPTION
**Pull request by omarudolley**
Implementation of the Manufacturing dataproducts definitions

### CODE REVIEW

File: src/DigitalProductPassport/Battery/ManufacturingDataSheet_v0.1.py [hunk 1]
```diff
@@ -0,0 +1,377 @@
...
+        ...,
+        title="Recycled Content",
+        description="The recycled content information present in the battery",
+    )
+    renewable_content: List[RenewableContent] = Field(
+        ...,
+        title="Renewable Content",
+        description="The renewable content information present in the battery",
+    )
+    legal_conformity: LegalConformity = Field(
```
Comment (joakimnordling) [inline src/DigitalProductPassport/Battery/ManufacturingDataSheet_v0.1.py:340]: `extinguishing_agent` was not supposed to be removed altogether, just changed to a list of strings.

File: src/DigitalProductPassport/Battery/HealthData_v0.1.py [hunk 2]
```diff
@@ -0,0 +1,179 @@
...
+    )
+    resistance_increase: float = Field(
+        ...,
+        title="Resistance Increase",
+        description="The value of resistance increase since the battery was first commissioned in percentage (%)",
+    )
+    operation_details: List[OperationDetail] = Field(
+        ...,
+        title="Operation Details",
+        description="The periodic information of he battery operation",
```
Comment (joakimnordling) [inline src/DigitalProductPassport/Battery/HealthData_v0.1.py:98]:
```suggestion
        description="The periodic information of the battery operation",
```

File: src/DigitalProductPassport/Battery/HealthData_v0.1.py [hunk 3]
```diff
@@ -0,0 +1,179 @@
...
+    )
+
+
+class HarmfulEvent(CamelCaseModel):
+    date: Optional[datetime] = Field(
```
Comment (joakimnordling) [inline src/DigitalProductPassport/Battery/HealthData_v0.1.py:103]: This should be a date, not a datetime.

File: src/DigitalProductPassport/Battery/HealthData_v0.1.py [hunk 4]
```diff
@@ -0,0 +1,179 @@
...
+    )
+    calendar_years: int = Field(
+        ...,
+        title="Calendar Years",
+        description="The expected lifetime of the battery in calendar years under the reference conditions for which it has been designed",
+    )
+
+
+class OperationDetail(CamelCaseModel):
+    date: datetime = Field(
```
Comment (joakimnordling) [inline src/DigitalProductPassport/Battery/HealthData_v0.1.py:50]: This should be a date, not a datetime.

File: src/DigitalProductPassport/MovableWorkMachine/EnvironmentalFootprint_v0.1.py
Comment (joakimnordling) [inline src/DigitalProductPassport/MovableWorkMachine/EnvironmentalFootprint_v0.1.py:1]: Supposed to be in MobileWorkMachine, not MovableWorkMachine

File: src/DigitalProductPassport/Battery/HealthData_v0.1.py [hunk 5]
```diff
@@ -0,0 +1,180 @@
...
+    )
+    calendar_years: Optional[int] = Field(
+        None,
+        title="Calendar Years",
+        description="The expected lifetime of the battery in calendar years under the reference conditions for which it has been designed",
+    )
+
+
+class OperationDetail(CamelCaseModel):
+    date: Optional[datetime] = Field(
```
Comment (joakimnordling) [inline src/DigitalProductPassport/Battery/HealthData_v0.1.py:51]: This one should as well be a date, not a datetime.

**Review by joakimnordling — COMMENTED** [review]
Apart from the fields that should be dates and not dates this looks good to me. There's however some issue with the tooling (openapi-to-fastapi and validation there) that causes issues for the pre-commit hooks if the fields are changed to dates, so this will require a bit of investigation of the issues with that tooling.

File: src/DigitalProductPassport/Battery/HealthData_v0.1.py [hunk 6]
```diff
@@ -0,0 +1,180 @@
+from datetime import datetime
+from enum import Enum
+from typing import List, Optional
+
+from definition_tooling.converter import CamelCaseModel, DataProductDefinition
+from pydantic import EmailStr, Field
```
Comment (joakimnordling) [inline src/DigitalProductPassport/Battery/HealthData_v0.1.py:6]: `EmailStr` is not used in this definition.

File: src/DigitalProductPassport/Battery/HealthData_v0.1.py (hunk 5 above)
Comment (joakimnordling) [reply src/DigitalProductPassport/Battery/HealthData_v0.1.py]: Renamed this one as well.

**Review by joakimnordling — APPROVED** [review]

**Review by hsuomi — CHANGES_REQUESTED** [review]

**Review by hsuomi — APPROVED** [review]

**Pull request merged**